### PR TITLE
NSErrorException should not accept null error, as it will throw later

### PR DESCRIPTION
**Pull request by chamons**
- https://bugzilla.xamarin.com/show_bug.cgi?id=42033
- Multiple proerties would poke the NSError reference, and NullReferenceException if a null was passed in.

### CODE REVIEW

**Comment by chamons** [comment]
@spouliot @rolfbjarne I know it's a trivial change, but is this safe enough for the iron clad backwards compatibility of iOS? 

Technically someone could have made these w/ null, never looked inside, and now we're throw "early".

**Comment by xamarin-release-manager** [comment]
Build success

File: src/Foundation/NSErrorException.cs [hunk 1]
```diff
@@ -31,6 +31,8 @@ public class NSErrorException : Exception {
 
 		public NSErrorException (NSError error)
 		{
+			if (error == null)
+				throw new System.ArgumentNullException ("error");
```
Comment (spouliot) [inline src/Foundation/NSErrorException.cs]: very minor: no need to prepend `System.` and a nice place to use C#6 `nameof`

**Comment by spouliot** [comment]
I'm fine with it. It's common that we add/remove null checks to avoid crashes or to allow API that accept it. 

Unless @alanmcgovern had a scenario which made sense to have null ?

**Comment by xamarin-release-manager** [comment]
Build failure

**Comment by alanmcgovern** [comment]
I don't see any case where passing in null would've resulted in good behaviour before :) If we're doing it accidentally we'll catch it now! I did review our usage and we did seem to be ok though.

**Comment by spouliot** [comment]
@chamons let's bring this into `cycle8` since this is where most of the short-term testing will be done.

**Comment by chamons** [comment]
@spouliot Card'ed for C8 https://trello.com/c/r3jY357e/79-xm-xi-nserrorexception-should-not-accept-null-error-as-it-will-throw-later as requested

**Pull request merged**